### PR TITLE
netfetch: init at 5.2.10

### PR DESCRIPTION
**Pull request by Banh-Canh**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ne/netfetch/package.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    mv $out/bin/backend $out/bin/$pname
+    installShellCompletion --cmd $pname \
+      --bash <($out/bin/$pname completion bash) \
+      --fish <($out/bin/$pname completion fish) \
+      --zsh <($out/bin/$pname completion zsh)
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/deggja/netfetch";
+    description = "Kubernetes tool for scanning clusters for network policies and identifying unprotected workloads.";
```
Comment (eclairevoyant) [inline pkgs/by-name/ne/netfetch/package.nix:41]:
```suggestion
    description = "Kubernetes tool for scanning clusters for network policies and identifying unprotected workloads";
```

File: pkgs/by-name/ne/netfetch/package.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  proxyVendor = true;
+  subPackages = [ "backend" ];
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/deggja/netfetch/backend/cmd.Version=${version}"
+  ];
+  nativeBuildInputs = [ installShellFiles ];
+  postInstall = ''
```
Comment (eclairevoyant) [inline pkgs/by-name/ne/netfetch/package.nix:31]:
```suggestion
  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform)  ''
```

File: pkgs/by-name/ne/netfetch/package.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    mv $out/bin/backend $out/bin/$pname
+    installShellCompletion --cmd $pname \
+      --bash <($out/bin/$pname completion bash) \
+      --fish <($out/bin/$pname completion fish) \
+      --zsh <($out/bin/$pname completion zsh)
+  '';
+
+  meta = with lib; {
```
Comment (eclairevoyant) [inline pkgs/by-name/ne/netfetch/package.nix:39]:
```suggestion
  meta = {
```

#292468

File: pkgs/by-name/ne/netfetch/package.nix [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    homepage = "https://github.com/deggja/netfetch";
+    description = "Kubernetes tool for scanning clusters for network policies and identifying unprotected workloads.";
+    license = licenses.mit;
+    mainProgram = "netfetch";
+    maintainers = with maintainers; [ banh-canh ];
```
Comment (eclairevoyant) [inline pkgs/by-name/ne/netfetch/package.nix:44]:
```suggestion
    license = lib.licenses.mit;
    mainProgram = "netfetch";
    maintainers = with lib.maintainers; [ banh-canh ];
```

File: pkgs/by-name/ne/netfetch/package.nix [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+  src = fetchFromGitHub {
+    owner = "deggja";
+    repo = "netfetch";
+    rev = "${version}";
+    hash = "sha256-N3wKpWdG92cXH0TwAkcsld9TRrfPRkbw0uZY/X4d+xk=";
+  };
+
+  vendorHash = "sha256-/Em3hx5tiQjThLBPJDHGsqxUV3eXeymJ5pY9c601OW0=";
+  proxyVendor = true;
```
Comment (eclairevoyant) [inline pkgs/by-name/ne/netfetch/package.nix:22]: why was this needed here?

**Review by eclairevoyant — DISMISSED** [review]

File: pkgs/by-name/ne/netfetch/package.nix (hunk 5 above)
Comment (Banh-Canh) [reply pkgs/by-name/ne/netfetch/package.nix:22]: I encountered this issue: https://github.com/NixOS/nixpkgs/issues/203039
Due to the main project making use of go workplaces.

File: pkgs/by-name/ne/netfetch/package.nix [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  fetchFromGitHub,
+  installShellFiles,
+  stdenv,
+}:
+let
+  version = "5.2.10";
+in
+buildGoModule {
+  pname = "netfetch";
+  version = "5.2.10";
```
Comment (eclairevoyant) [inline pkgs/by-name/ne/netfetch/package.nix:13]: I'd use either let or rec here, but writing out version twice doesn't make sense IMO

Comment (Banh-Canh) [reply pkgs/by-name/ne/netfetch/package.nix]: Opted for rec version, i like it, less verbosity but i'm open to anything
